invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/search/DeleteModal.js

```diff
@@ -0,0 +1,141 @@
+// This file is part of InvenioAdministration
+// Copyright (C) 2022 CERN.
+//
+// Invenio RDM Records is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Button, Icon, Message, Checkbox } from "semantic-ui-react";
+import { InvenioAdministrationActionsApi } from "@js/invenio_administration/api/actions";
+import { i18next } from "@translations/invenio_administration/i18next";
+import { Modal } from "semantic-ui-react";
+import { Trans } from "react-i18next";
+import isEmpty from "lodash/isEmpty";
+import { ErrorMessage } from "@js/invenio_administration/ui_messages/messages";
+import { NotificationContext } from "@js/invenio_administration/ui_messages/context";
+
+export class DeleteModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { loading: false, error: undefined, checked: false };
+  }
+
+  static contextType = NotificationContext;
+
+  cleanError = () => {
+    this.setState({ error: undefined });
+  };
+
+  handleOnButtonClick = async () => {
+    const { successCallback, resource, apiEndpoint, idKeyPath, toggleModal } =
+      this.props;
+    const { addNotification } = this.context;
+    try {
+      await InvenioAdministrationActionsApi.deleteResource(
+        resource,
+        apiEndpoint,
+        idKeyPath
+      );
+      toggleModal(false);
+      addNotification({
+        title: "Success",
+        content: "Resource was successfully deleted.",
+        type: "success",
+      });
+      successCallback();
+    } catch (e) {
+      this.setState({
+        error: { header: "Action error", content: e, id: e.code },
+      });
+    }
+  };
```

Comment:
do you need to copy this part with using overridable? in this way you have this function defined twice, since you are only placing overridable ID in the render function of the overridden component

Reply:
RIght! Good point, thanks!